File: betterrandom/src/main/java/io/github/pr0methean/betterrandom/SeekableRandom.java
package io.github.pr0methean.betterrandom;

/**
 * A {@link RepeatableRandom} that can skip backward or forward within its sequence of output.
 */
public interface SeekableRandom extends RepeatableRandom {

  /**
   * Advances the generator forward {@code delta} steps, but does so in logarithmic time.
   *
   * @param delta the number of steps to advance; can be negative
   */
  void advance(long delta);
}

File: betterrandom/src/main/java/io/github/pr0methean/betterrandom/SeekableRandom.kt
package io.github.pr0methean.betterrandom

interface SeekableRandom : RepeatableRandom {

    /**
     * Advances the generator forward `delta` steps, but does so in logarithmic time.
     * @param delta the number of steps to advance; can be negative
     */
    fun advance(delta: Long)
}
